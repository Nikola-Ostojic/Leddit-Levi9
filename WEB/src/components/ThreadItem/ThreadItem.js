import React from "react";
import { Card } from "@material-ui/core";

import styles from "./ThreadItem.module.css";

const ThreadItem = ({ thread, onClick }) => (
  <Card
    className={styles.container}
    raised
    onClick={() => onClick(`/threadDetails/${thread.id}`)}
  >
    <div className={styles.headerContainer}>
      <h2>{thread.title}</h2>
      <h2>{thread.author}</h2>
    </div>
    <div className={styles.content}>{thread.content}</div>
    <div className={styles.footerContainer}>
      <h3>Comments {thread.commentsCount}</h3>
    </div>
  </Card>
);

export default ThreadItem;
