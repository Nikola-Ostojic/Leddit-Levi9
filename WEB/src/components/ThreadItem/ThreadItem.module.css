.container {
    margin-top: 10px;
    padding: 15px;
    box-sizing: border-box;
}

.headerContainer {
    width: 100%;
    display: flex;
    justify-content: space-between;
}

.content {
    margin-top: 20px;
    margin-bottom: 20px;
}

.footerContainer {
    width: 100%;
    display: flex;
    justify-content: space-between;
}