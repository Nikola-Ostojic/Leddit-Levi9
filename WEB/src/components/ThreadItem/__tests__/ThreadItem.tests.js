import React from "react";
import { shallow } from "enzyme";

import { Card } from "@material-ui/core";

import ThreadItem from "../ThreadItem";

describe("ThreadItem component tests", () => {
  const mockedProps = {
    thread: {
      title: "Test",
      content: "Test"
    },
    onClick: jest.fn()
  };
  it("Should match snapshot", () => {
    const wrapper = shallow(<ThreadItem {...mockedProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should call on click when item is clicked", () => {
    const wrapper = shallow(<ThreadItem {...mockedProps} />);

    wrapper.find(Card).simulate("click");

    expect(mockedProps.onClick).toHaveBeenCalled();
  });
});
