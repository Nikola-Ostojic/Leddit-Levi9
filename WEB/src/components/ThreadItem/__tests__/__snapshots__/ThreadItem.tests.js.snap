// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadItem component tests Should match snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="container"
  onClick={[Function]}
  raised={true}
>
  <div
    className="headerContainer"
  >
    <h2>
      Test
    </h2>
    <h2 />
  </div>
  <div
    className="content"
  >
    Test
  </div>
  <div
    className="footerContainer"
  >
    <h3>
      Comments 
    </h3>
  </div>
</WithStyles(ForwardRef(Card))>
`;
